### ISSUE TITLE
FR: more options for deate formats

### ISSUE THREAD

**Issue by CarnageMarkus** [mainsail-crew/mainsail#2038]
### Requested feature:

While I am not native English speaker, I am used to english terminology in tech and therefore use English everywhere except for when it comes to date and time formats. I would like to see those in "regional format". Windows has allowed this for ages with web it's little difficult.

For some reason I am only able to pick from these date formats (MDY)
![image](https://github.com/user-attachments/assets/9c434238-b4b0-4371-8d39-bdaae89e6428)

This as well might be "bug" because I tried setting english-US (MDY) and english-GB (DMY) as language in my browser (Firefox)
I tried executing ```alert(new Date().toLocaleString())``` in console which correctly shows time in DMY format
![image](https://github.com/user-attachments/assets/5239203b-c9a9-4c11-aacd-53fd34330ff5)
So it looks like bowser locale detection is not working properly ?

TLDR: I think it should be possible to choose whatever time format for english as many non native speakers are used to see interfaces in english while preferring to see dates in DMY, MDY or YMD, just like time can be in 12/24 

Thank you. ❤️
